### ISSUE TITLE
AI Assistant: Add upgrade banner to block whenever user runs out of requests

### ISSUE THREAD

**Issue by nunyvega** [Automattic/jetpack#30736]
Blocked by [#30726](https://github.com/Automattic/jetpack/issues/30726), we need to build the requests counter first. 

Once the user runs out of requests, add a way to let the user upgrade or buy more credits. 

As a first iteration, maybe we can use something similar to what we have in VideoPress, but with a link to the interstitial for Jetpack AI:

<img width="629" alt="Screenshot 2023-05-16 at 11 36 57" src="https://github.com/Automattic/jetpack/assets/16329583/9fa224e6-30aa-4709-84b5-d860072aa68c">

**Comment by keoshi** [comment]
Here's a couple of quick mocks of how this information could be presented:

1. This one uses a similar upgrade style as premium blocks in Dotcom
![image](https://github.com/Automattic/jetpack/assets/390760/e5e48790-18b0-49e6-8650-53c8590f79f8)

gG1smMHbNGas0qwP9y6RwK-fi-3609_17698

2. This one uses a slightly subtler style but more contextual

![image](https://github.com/Automattic/jetpack/assets/390760/7c7a1147-446e-4e61-9d8c-89a1a4215159)

gG1smMHbNGas0qwP9y6RwK-fi-3609_17699

**Comment by retrofox** [comment]
It

**Comment by retrofox** [comment]
closing in favor of https://github.com/Automattic/jetpack/issues/30842
